crates/utils/src/errors.rs

```diff
@@ -0,0 +1,14 @@
+#[derive(Debug)]
+pub enum ApiError {
+    ApiInitialisationFailed(String),
+    ApiServeFailed(String),
+    AddressResolutionFailed(String),
```

Comment:
Currently this `String` is used only for error message about address resolution fail. It would be better to not to provide `String` here, but write error message for this variant in `Display` implementation.

Reply:
The issue is still here. It would be better to just remove this `String` argument.